src/test/java/io/gravitee/common/utils/DurationParserTest.java

```diff
@@ -0,0 +1,124 @@
+
+    @ParameterizedTest
+    @MethodSource("provideIso8601Duration")
+    void shouldReturnDurationWhenValueIsIso8601(final String given, final String expected) {
+        assertThat(DurationParser.parse(given)).isEqualTo(Duration.parse(expected));
+    }
+
+    private static Stream<Arguments> provideSimpleDuration() {
+        return Stream.of(
+            // Simple duration / days / hours / minutes / sec / millis
+
+            Arguments.of("10d", 10, 240, 14400, 864000, 864000000),
+            Arguments.of("10D", 10, 240, 14400, 864000, 864000000),
+            Arguments.of("+10d", 10, 240, 14400, 864000, 864000000),
+            Arguments.of("-10d", -10, -240, -14400, -864000, -864000000),
+            Arguments.of("10h", 0, 10, 600, 36000, 36000000),
+            Arguments.of("10H", 0, 10, 600, 36000, 36000000),
+            Arguments.of("+10h", 0, 10, 600, 36000, 36000000),
+            Arguments.of("-10h", 0, -10, -600, -36000, -36000000),
+            Arguments.of("10m", 0, 0, 10, 600, 600000),
+            Arguments.of("10M", 0, 0, 10, 600, 600000),
+            Arguments.of("+10m", 0, 0, 10, 600, 600000),
+            Arguments.of("-10m", 0, 0, -10, -600, -600000),
+            Arguments.of("10s", 0, 0, 0, 10, 10000),
+            Arguments.of("10S", 0, 0, 0, 10, 10000),
+            Arguments.of("+10s", 0, 0, 0, 10, 10000),
+            Arguments.of("-10s", 0, 0, 0, -10, -10000),
+            Arguments.of("10ms", 0, 0, 0, 0, 10),
+            Arguments.of("10MS", 0, 0, 0, 0, 10),
+            Arguments.of("+10ms", 0, 0, 0, 0, 10),
+            Arguments.of("-10ms", 0, 0, 0, -1, -10)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideSimpleDuration")
+    void shouldReturnDurationWhenValueIsMillis(
+        final String given,
+        final long expectedDays,
+        final long expectedHours,
+        final long expectedMinutes,
+        final long expectedSeconds,
+        final long expectedMillis
+    ) {
+        assertThat(DurationParser.parse(given))
+            .hasDays(expectedDays)
+            .hasHours(expectedHours)
+            .hasMinutes(expectedMinutes)
+            .hasSeconds(expectedSeconds)
+            .hasMillis(expectedMillis);
+    }
+
+    private static Stream<Arguments> provideBadFormatSimpleDuration() {
+        return Stream.of(
+            Arguments.of(" "),
+            Arguments.of("1 "),
+            Arguments.of("1"),
+            Arguments.of("1d "),
+            Arguments.of("1h "),
+            Arguments.of("1m "),
+            Arguments.of("1s "),
+            Arguments.of("1ms "),
+            Arguments.of("1md"),
+            Arguments.of("1pou")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideBadFormatSimpleDuration")
+    void shouldReturnNullWhenValueHasBadFormat(final String given) {
+        Duration duration = DurationParser.parse(given);
+        assertThat(duration).isNull();
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
```

Comment:
If I remember correctly, you can add several source annotations on a test. So if you add `@NullAndEmptySource` in the previous test case, we can have only one method checking bad format.